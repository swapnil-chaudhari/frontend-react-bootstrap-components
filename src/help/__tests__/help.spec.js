import Help from '../help';
import React from 'react';
import { Overlay } from 'react-bootstrap';
import renderShallow from 'render-shallow';
import { findWithType } from 'react-shallow-testutils';

describe('<Help>', () => {
    describe('when rendered with required props', () => {
        const props = {
            message: 'Here is your help',
            id: 'help-foo',
            tooltipPlacement: 'top'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Help { ...props } />).output;
        });

        test('renders <Help> with the right props', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with children', () => {
        const props = {
            message: 'Hello World',
            id: 'hello-world',
            tooltipPlacement: 'right',
            children: <span>Hello World</span>,
            rootClose: true
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Help { ...props } />).output;
        });

        test('renders <Help> with children', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with a popover overlay type', () => {
        const props = {
            message: 'Hello World',
            id: 'hello-world',
            tooltipPlacement: 'bottom',
            overlayType: 'popover',
            rootClose: false
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Help { ...props } />).output;
        });

        test('renders <Help> with a <Popover />', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with a message node', () => {
        const props = {
            message: <span>'Hello World'</span>,
            id: 'hello-world',
            tooltipPlacement: 'bottom',
            rootClose: false
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Help { ...props } />).output;
        });

        test('renders <Help> with a <Popover />', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with an overlay class name', () => {
        const props = {
            message: 'Hello World',
            id: 'hello-world',
            overlayClassName: 'overlayClassName',
            tooltipPlacement: 'bottom',
            rootClose: false
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Help { ...props } />).output;
        });

        test('renders a <Tooltip /> in <Help /> with a custom class', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with triggerOverride = show', () => {
        const props = {
            message: 'Hello World',
            id: 'hello-world',
            triggerOverride: 'show'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Help { ...props } />).output;
        });

        test('renders an <Overlay /> that is always visible', () => {
            const overlayProps = findWithType(component, Overlay).props;
            expect(overlayProps.show).toBeTruthy();
        });
    });

    describe('when rendered with triggerOverride = hide', () => {
        const props = {
            message: 'Hello World',
            id: 'hello-world',
            children: <span>Hello World</span>,
            triggerOverride: 'hide'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Help { ...props } />).output;
        });

        test('renders an <Overlay /> that is always hidden', () => {
            const overlayProps = findWithType(component, Overlay).props;
            expect(overlayProps.show).toBeFalsy();
        });
    });
});
