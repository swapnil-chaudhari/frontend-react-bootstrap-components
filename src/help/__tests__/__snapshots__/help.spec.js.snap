// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Help> when rendered with a message node renders <Help> with a <Popover /> 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Tooltip
      bsClass="tooltip"
      className={undefined}
      id="hello-world"
      placement="right"
    >
      <span>
        'Hello World'
      </span>
    </Tooltip>
  }
  placement="bottom"
  rootClose={false}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span>
    <Info />
  </span>
</OverlayTrigger>
`;

exports[`<Help> when rendered with a popover overlay type renders <Help> with a <Popover /> 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className={undefined}
      id="hello-world"
      placement="right"
    >
      Hello World
    </Popover>
  }
  placement="bottom"
  rootClose={false}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span>
    <Info />
  </span>
</OverlayTrigger>
`;

exports[`<Help> when rendered with an overlay class name renders a <Tooltip /> in <Help /> with a custom class 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Tooltip
      bsClass="tooltip"
      className="overlayClassName"
      id="hello-world"
      placement="right"
    >
      Hello World
    </Tooltip>
  }
  placement="bottom"
  rootClose={false}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span>
    <Info />
  </span>
</OverlayTrigger>
`;

exports[`<Help> when rendered with children renders <Help> with children 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Tooltip
      bsClass="tooltip"
      className={undefined}
      id="hello-world"
      placement="right"
    >
      Hello World
    </Tooltip>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span>
    Hello World
  </span>
</OverlayTrigger>
`;

exports[`<Help> when rendered with required props renders <Help> with the right props 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Tooltip
      bsClass="tooltip"
      className={undefined}
      id="help-foo"
      placement="right"
    >
      Here is your help
    </Tooltip>
  }
  placement="top"
  rootClose={false}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span>
    <Info />
  </span>
</OverlayTrigger>
`;
